src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -136,7 +137,16 @@ public static async Task LoadAndRunInteractiveDocument(
                 {
                     submitCode.SetParent(parentCommand);
                 }
-                await kernel.RootKernel.SendAsync(submitCode);
+
+                var result = await kernel.RootKernel.SendAsync(submitCode);
+
+                // FIX: (LoadAndRunInteractiveDocument) consider publishing completion events only visible to the inner command so that he API doesn't have this difference in behavior.
+                if (result.Events.Count == 0)
+                {
```

Comment:
nit: Consolidate the condition to avoid the empty block?

Reply:
Removing this entirely. It was only here for code coverage feedback.